### PR TITLE
Add docs on how to use emscripten in GitLab

### PR DESCRIPTION
**Pull request by oxr463**
This PR relies on the following issue:

-  https://gitlab.com/pages/pages.gitlab.io/issues/75

### CODE REVIEW

**Comment by kripken** [comment]
Thanks, good idea!

This has some errors in the build-docs job which we need to fix. (We do need ci even for docs changes.)

Also need to merge in latest incoming to here, to fix the other CI issues.

**Comment by oxr463** [comment]
- [ ] `/root/project/site/source/docs/compiling/GitLab.rst:3:Title overline too short.`

File: site/source/docs/compiling/GitLab.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+This step also contains the command to add an additional build tool *make*.
+
+.. code-block:: yaml
+
+    script:
+      - make
+
+In the script stage we can now run the commands we want. In this sample we are using *make*, but you can call *emcc* directly if you prefer.
+
+For an example of this setup in practice, see `the Example Emscripten site using GitLab Pages <https://pages.gitlab.io/emscripten>`_.
```
Comment (sbc100) [inline site/source/docs/compiling/GitLab.rst]: This link doesn't seem to work.

Comment (oxr463) [reply site/source/docs/compiling/GitLab.rst]: As stated in the description, this PR is pending based on the link above.

I'll clarify further, by stating that the official GitLab Pages example url won't work until after the PR for GitLab Pages is merged.

**Comment by kripken** [comment]
This is failing the build-docs CI task, we need to fix that before landing.

**Comment by kripken** [comment]
Perfect, thanks!

**Comment by oxr463** [comment]
@kripken, I was leaving this as WIP until GitLab accepted my other PR.

As pointed out by @sbc100, the GitLab Pages URL won't be live until then.

**Comment by kripken** [comment]
Oh, sorry, should I revert this?

**Comment by oxr463** [comment]
I don't like the idea of leaving that link dangling like that.

We could just replace it with my demo repository until it gets merged, e.g., `https://pages.gitlab.io/emscripten -> https://lramage.gitlab.io/emscripten`.

**Comment by kripken** [comment]
It won't go onto the live website until we update it anyhow. I'll hold off until we see that other one merged. Do you have some guess as to how long that might take?

**Comment by oxr463** [comment]
Honestly, I have no idea. I haven't even gotten any comments on it yet.

**Comment by gtsiolis** [comment]
Thanks @oxr463 @kripken! I've [pinged](https://gitlab.com/pages/pages.gitlab.io/issues/75#note_188708129) someone on the relevant issue.

**Comment by oxr463** [comment]
FYI, @kripken, the URL is live, https://pages.gitlab.io/emscripten

**Comment by kripken** [comment]
Thanks @oxr463 - I opened #8944, please check I did it ok.

**Pull request merged**